FirebaseMessaging/Tests/IntegrationTests/FIRMessagingPubSubTest.swift

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// macOS requests a user password when accessing the Keychain for the first time,
+// so the tests may fail. Disable integration tests on macOS so far.
+// TODO: Configure the tests to run on macOS without requesting the keychain password.
+#if !os(OSX)
+
+  import FirebaseCore
+  import FirebaseMessaging
+  import XCTest
+
+  class FIRMessagingPubSubTest: XCTestCase {
+    var app: FirebaseApp!
+    var messaging: Messaging?
+
+    override class func setUp() {
+      FirebaseApp.configure()
+    }
+
+    override func setUp() {
+      messaging = Messaging.messaging()
+    }
+
+    override func tearDown() {
+      messaging = nil
+    }
+
+    func testSubscribeTopic() {
+      let expectation = self.expectation(description: "Successfully subscribe topic")
+      assertDefaultToken()
+
+      guard let messaging = self.messaging else {
+        return
+      }
+      messaging.subscribe(toTopic: "cat_video") { error in
+        XCTAssertNil(error)
+        expectation.fulfill()
+      }
+      wait(for: [expectation], timeout: 5)
+    }
+
+    func testUnsubscribeTopic() {
+      let expectation = self.expectation(description: "Successfully unsubscribe topic")
+      assertDefaultToken()
+
+      guard let messaging = self.messaging else {
+        return
+      }
+      messaging.unsubscribe(fromTopic: "cat_video") { error in
+        XCTAssertNil(error)
+        expectation.fulfill()
+      }
+      wait(for: [expectation], timeout: 5)
+    }
+
+    func assertDefaultToken() {
```

Comment:
Please add the async variation of the test like `testScalar` and `testScalarAsync` at https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseFunctions/Tests/SwiftIntegration/IntegrationTests.swift#L97

Reply:
> 



> Good progress!
> 
> Here's an example of setting up a GoogleService-Info.plist for integration tests - https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/build.sh#L528

@paulb777 Is the build script for presubmit tho? The plist file is not associated with test bundle locally when running  pod gen FirebaseMessaging.podspec

Reply:
@chliangGoogle For running locally, you either need to run the ruby command or manually add `GoogleService-Info.plist` to the  AppHost-FirebaseStorageMessaging... target in the Xcode UI

Reply:
@paulb777 Can you give more context on the async vatiation test? I do have async tests in integration. What are the differences here?

Reply:
There should be tests to verify the auto-generated `async` versions of the APIs that have a completion handler like the difference between the `testScalar` and `testScalarAsync` linked above.

Reply:
Oh this is for the async/await! Gotcha!